tests/subsys/mpsl/pm/pm_test.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+
+#include <errno.h>
+
+#include "cmock_policy.h"
+#include "cmock_mpsl_pm.h"
+#include "cmock_mpsl_pm_config.h"
+
+#include <mpsl/mpsl_pm_utils.h>
+#include <zephyr/kernel.h>
+
+#define PM_MAX_LATENCY_HCI_COMMANDS_US 4999999
+#define MAX_DELAY_SINCE_READING_PARAMS 50
+
+/* The unity_main is not declared in any header file. It is only defined in the generated test
+ * runner because of ncs' unity configuration. It is therefore declared here to avoid a compiler
+ * warning.
+ */
+extern int unity_main(void);
+
+void m_pm_init(mpsl_pm_params_t *p_test_params)
+{
+	__cmock_pm_policy_latency_request_add_Expect(0, PM_MAX_LATENCY_HCI_COMMANDS_US);
+	__cmock_pm_policy_latency_request_add_IgnoreArg_req();
+
+	__cmock_mpsl_pm_init_Expect();
+
+	__cmock_mpsl_pm_params_get_ExpectAnyArgsAndReturn(true);
+	__cmock_mpsl_pm_params_get_ReturnThruPtr_p_params(p_test_params);
+
+	pm_init();
+}
+
+void m_pm_work_handling_update_latency(mpsl_pm_params_t *p_test_params, bool exp_lat_update)
+{
+	p_test_params->cnt_flag++;
+
+	__cmock_mpsl_pm_params_get_ExpectAnyArgsAndReturn(true);
+	__cmock_mpsl_pm_params_get_ReturnThruPtr_p_params(p_test_params);
+
+	if (exp_lat_update) {
+		if (p_test_params->event_state == MPSL_PM_EVENT_STATE_IN_EVENT) {
+			__cmock_pm_policy_latency_request_update_Expect(0, 0);
+		} else {
+			__cmock_pm_policy_latency_request_update_Expect(0,
+						PM_MAX_LATENCY_HCI_COMMANDS_US);
+		}
+		__cmock_pm_policy_latency_request_update_IgnoreArg_req();
+	}
+	pm_work_handling();
+}
+
+void m_pm_work_handling_event_register(mpsl_pm_params_t *p_test_params, bool exp_event_registered,
+					uint64_t test_rel_evt_time_us)
+{
+	__cmock_mpsl_pm_params_get_ExpectAnyArgsAndReturn(true);
+	__cmock_mpsl_pm_params_get_ReturnThruPtr_p_params(p_test_params);
+
+	p_test_params->cnt_flag++;
+	p_test_params->event_time_rel_us = test_rel_evt_time_us;
+
+	while ((test_rel_evt_time_us - MAX_DELAY_SINCE_READING_PARAMS) > UINT32_MAX) {
+		__cmock_mpsl_pm_params_get_ExpectAnyArgsAndReturn(true);
+		__cmock_mpsl_pm_params_get_ReturnThruPtr_p_params(p_test_params);
+		p_test_params->event_time_rel_us = p_test_params->event_time_rel_us -
+						   MAX_DELAY_SINCE_READING_PARAMS;
+	}
```

Comment:
Because I am testing PM and using the real k_work stuffs, it would reenter (or rather call again) pm_work_handling in case of delayed handling. Since the UT would fail that expectations aren't met I assumed that this is actually how it is also working.

Reply:
The tests pass if these lines are deleted, too.